src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/ConfigurationFilesExternalSettingsProviderService.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            SettingValuesChanged?.Invoke(this, ExternalSettingsChangedEventArgs.SomeOrAll);
+        }
+
+        public event EventHandler<ExternalSettingsChangedEventArgs>? SettingValuesChanged;
+
+        // Event is unused at this time, so an empty add and remove accessor block is used to avoid a CS0067 analyzer warning.
+        public event EventHandler<EnumSettingChoicesChangedEventArgs> EnumSettingChoicesChanged { add { } remove { } }
+        // Event is unused at this time, so an empty add and remove accessor block is used to avoid a CS0067 analyzer warning.
+        public event EventHandler<DynamicMessageTextChangedEventArgs> DynamicMessageTextChanged { add { } remove { } }
+        // Event is unused at this time, so an empty add and remove accessor block is used to avoid a CS0067 analyzer warning.
+        public event EventHandler ErrorConditionResolved { add { } remove { } }
+
+        public void Dispose()
+        {
+            if (_vsSettings != null)
+            {
+                _vsSettings.SettingsChanged -= VsSettings_SettingsChanged;
+            }
+        }
+
+        public Task<ExternalSettingOperationResult<IReadOnlyList<EnumChoice>>> GetEnumChoicesAsync(string enumSettingMoniker, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<string> GetMessageTextAsync(string messageId, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<ExternalSettingOperationResult<T>> GetValueAsync<T>(string moniker, CancellationToken cancellationToken) where T : notnull
+        {
+            switch (moniker)
+            {
+                case MonikerConfigurationFiles: return LoadConfigurationFilePathsOrThrow<T>(_settings!);
+                default: break;
+            }
+
+            // Shouldn't happen as these are monikers we declared in registration.json.
+            throw new InvalidOperationException();
+        }
+
+        public Task OpenBackingStoreAsync(CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<ExternalSettingOperationResult> SetValueAsync<T>(string moniker, T value, CancellationToken cancellationToken) where T : notnull
+        {
+            throw new NotImplementedException();
+        }
+
+        private static Task<ExternalSettingOperationResult<T>> LoadConfigurationFilePathsOrThrow<T>(ISettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            ExternalSettingOperationResult<T> result;
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                var configPathsList = settings.GetConfigFilePaths().ToList();
```

Comment:
This has a minor perf impact here, so it's not super important, but why ToList?
It's already an IList.

Reply:
Great point. I just copied our existing logic and didn't pay much attention to that.